assets/js/lesson_5/hw_10.js

```diff
@@ -0,0 +1,230 @@
+    taskFour += i
+}
+
+console.log(`Answer is ${taskFour}`)
+
+// Added prompt just for fun
+
+let num = prompt('Enter whole number')
+let taskFourVar = 0
+const check = num < 0
+
+if (check) {
+   while (check) {
+    num = prompt('Try again')
+   }
+} else {
+    for (i = 1; i <= num; i++) {
+        taskFourVar += i
+    }
+}
+
+console.log(`Result is ${taskFourVar}`)
+
+
+/**
+ *  5. Find product of numbers from 15 to 35
+ *  We can use here prompt to enter a number, and check
+ *  if result is BigInt. If true we show result as bigint,
+ *  if false as ordinary number. Am I right?
+ * */
+
+let taskFive = 1;
+
+for (i = 15; i <= 35; i++) {
+    taskFive *= i;
+}
+
+console.log(BigInt(taskFive))
+
+
+// 6.Find average of numbers from 1 to 500
+
+let taskSix = 0;
+
+for (i = 1; i <= 500; i++) {
+    taskSix += i
+}
+
+const taskSixAverage = taskSix / 500
+
+console.log(taskSixAverage)
+
+// More complex solution of task 6
+
+// Get user input
+
+let taskSixValue = Number(prompt('Enter first value'))
+
+while (!taskSixValue || taskSixValue <= 0) {
+    taskSixValue = Number(prompt('Enter valid value'))
+}
+
+let taskSixSecValue = Number(prompt('Enter second value'))
+
+while (!taskSixSecValue || taskSixSecValue <= 0) {
+    taskSixSecValue = Number(prompt('Enter valid value'))
+}
+
+// Calculate average
+
+let taskSixArr = []
+
+for (let i = taskSixValue; i <= taskSixSecValue; i++) {
+    taskSixArr.push(i);
+}
+
+let taskSixRes = taskSixArr.reduce((a, b) => a + b, 0) / taskSixArr.length;
+
+alert(`Average of ${taskSixValue} and ${taskSixSecValue} = ${taskSixRes}`)
+
+
+// 7. Find sum of even numbers between 30 and 80
+
+let taskSeven = 0;
+
+for (i = 30; i <= 80; i++) {
+    if (i % 2 === 0) {
+        taskSeven += i
+    }
+}
+
+console.log(taskSeven)
+
+
+// 8. Show all numbers between 100 and 200 multiple of 3
+
+let taskEight = [];
+
+for (i = 100; i <= 200; i++) {
+    if (i % 3 === 0) {
+        taskEight.push(i)
+    }
+}
+
+console.log(taskEight.join(","))
+
+
+/** 
+ * 9, 10, 11. Find all dividers of whole number. Calculate quanity
+ * of even dividers and calculate their sum.
+ * */
+
+let wholeNum = Number(prompt('Enter the number'))
+
+while (!wholeNum || wholeNum <= 0) {
+    wholeNum = Number(prompt("Enter valid number! I'm begging you!"))
+}
+
+let isDivider = []
+let evenDividers = [];
+let dividerSum = 0;
+
+for (i = 1; i <= wholeNum; i++) {
+    if (wholeNum % i === 0) {
+        isDivider.push(i)
+
+        if (i % 2 === 0) {
+            evenDividers.push(i)
+            dividerSum += i
+        }
+    }
+}
+
+alert(`You've entered ${wholeNum} \n
+It has this dividers: ${isDivider} \n
+Several of them are even: ${evenDividers} \n
+Sum of even dividers is ${dividerSum}`)
+
+
+// 12. Print multiple table from 1 to 10
+
+const startTable = 1;
+const endTable = 10;
+
+for (let i = startTable; i <= endTable; i++) {
+    for (let j = startTable; j <= endTable; j++) {
+        console.log(`${i} * ${j} = ${i * j}`);
+    }
+}
```

Comment:
```suggestion
const startTable = 1;
const endTable = 10;

let result = [];

for (let i = startTable; i <= endTable; i++) {
  let iArray = [];
  for (let j = startTable; j <= endTable; j++) {
    iArray.push(`${i} * ${j} = ${i * j}`);
  }
  result.push(iArray);
}

console.table(result);
```

Reply:
If you want a table))

Reply:
Need to understand this tricks with array in 'for' loop inside 'for'. Thanks